client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -107,13 +107,19 @@ module.exports = React.createClass( {
 	},
 
 	getCheckoutCompleteRedirectPath: function() {
-		var renewalItem;
+		var renewalItem, path;
 		if ( cartItems.hasRenewalItem( this.props.cart ) ) {
 			renewalItem = cartItems.getRenewalItems( this.props.cart )[ 0 ];
-			return purchasePaths.managePurchaseDestination( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId, 'thank-you' );
+			path = purchasePaths.managePurchaseDestination( renewalItem.extra.purchaseDomain, renewalItem.extra.purchaseId, 'thank-you' );
+		} else {
+			path = '/checkout/thank-you';
 		}
 
-		return '/checkout/thank-you';
+		if ( this.props.cart.products.length === 1 && this.props.cart.products[0].free_trial ) {
+			path += '?free-trial=1';
```

Comment:
I grep'ed the branch a bit, and searched the diff for "query" or "param", but I can't figure out what the purpose of this querystring is. Can you leave a comment about what it does here (if it is something we want to keep)?